arbitrum-docs/devs-how-tos/how-to-estimate-gas.mdx

```diff
@@ -0,0 +1,142 @@
+    - For Arbitrum Nova (AnyTrust), the size of the data is also a fixed value, as only the Data Availability Certificate is posted on L1, [as explained here](/inside-anytrust#data-availability-certificates).
+
+## An example of how to apply this formula in your code
+
+Finally, we show an example of how to get the values we just described and how to estimate the gas usage of a transaction in Javascript. We will use our [SDK](https://github.com/OffchainLabs/arbitrum-sdk) to connect to `ArbGasInfo` and `NodeInterface`.
+
+We first instantiate a factory object for ArbGasInfo and NodeInterface, using two methods from the SDK. `l2Provider` is a regular JSON RPC provider for the L2 network we are using, and both `ARB_GAS_INFO` and `NODE_INTERFACE_ADDRESS` are the addresses of the precompiles in said network.
+
+```ts
+const { ArbGasInfo__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/ArbGasInfo__factory");
+const { NodeInterface__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/NodeInterface__factory");
+const { ARB_GAS_INFO, NODE_INTERFACE_ADDRESS } = require("@arbitrum/sdk/dist/lib/dataEntities/constants");
+
+...
+
+// Instantiation the ArbGasInfo and NodeInterface objects
+const arbGasInfo = ArbGasInfo__factory.connect(
+    ARB_GAS_INFO,
+    l2Provider
+);
+const nodeInterface = NodeInterface__factory.connect(
+    NODE_INTERFACE_ADDRESS,
+    l2Provider
+);
+```
+
+For this example, we will use the method `ArbGasInfo.getPricesInWei()` and `NodeInterface.gasEstimateComponents()` to get all the information we need. For the gasEstimateComponents() call we will pass a destinationAddress (might be any address) and the data we want to send, to get results as accurate as possible.
+
+```ts
+// Getting the gas prices from ArbGasInfo.getPricesInWei()
+const gasComponents = await arbGasInfo.callStatic.getPricesInWei();
+
+// And the estimations from NodeInterface.GasEstimateComponents()
+const gasEstimateComponents = await nodeInterface.callStatic.gasEstimateComponents(
+    destinationAddress,
+    false,
+    txData
+);
+```
+
+With this, we can now get the values of the 4 variables we will use in our formula.
+
+```ts
+// Setting the variables of the formula
+const P = gasComponents[5];
+const L2G = gasEstimateComponents[0].sub(gasEstimateComponents[1]);
+const L1P = gasComponents[1];
+const L1S = 140 + txDataLengthInBytes;
+```
+
+And finally, we estimate the transaction fees applying the formula described in the beginning.
+
+```ts
+// L1C (L1 Cost) = L1P * L1S
+const L1C = L1P.mul(L1S);
+
+// B (Extra Buffer) = L1C / P
+const B = L1C.div(P);
+
+// G (Gas Limit) = L2G + B
+const G = L2G.add(B);
+
+// TXFEES (Transaction fees) = P * G
+const TXFEES = P.mul(G);
+```
+
+You can also see a working example of this code in [our tutorials repository](https://github.com/OffchainLabs/arbitrum-tutorials/tree/master/packages/gas-estimation).
```

Comment:
```suggestion
Refer to [our tutorials repository](https://github.com/OffchainLabs/arbitrum-tutorials/tree/master/packages/gas-estimation) for a working example of this code.
```

Editing for succinctness. We'll want to update this to refer to the repo as the `Samples repository` when we make that change.